Automattic/sensei
# Base for course outline content

### Changes proposed in this Pull Request

* This PR introduces the base for the course outline content.
* It's getting the content from a mocked array (in both frontend and editor) until we have the API.
* This PR removes the legacy list when the course outline block is rendering. But this approach is temporary until we research the best approach.
* This PR still doesn't handle the styles, different lessons statuses, and other specific interactions.

### Testing instructions

* Create a new course.
* Interact with the outline block to make sure everything is working properly.
* Save the course with the outline block.
* Go to the frontend and make sure the course appears correctly (the data won't still match the correct content, it's getting a mocked array until we have the API).


### Screenshot / Video

Editor:

<img width="659" alt="Screen Shot 2020-09-03 at 18 06 06" src="https://user-images.githubusercontent.com/876340/92167140-456ed400-ee10-11ea-92ee-e2ab38adb671.png">

---

Frontend:

<img width="646" alt="Screen Shot 2020-09-03 at 18 06 30" src="https://user-images.githubusercontent.com/876340/92167148-4869c480-ee10-11ea-9cb4-f8ca0b1b1d83.png">
